auto_submit/test/requests/check_pull_request_test.dart

```diff
@@ -25,7 +25,7 @@ void main() {
       checkPullRequest = CheckPullRequest(config: config);
     });
 
-    test('call checkPullRequest handler to handle the get request', () async {
+    test('Merge PR with successful status and checks', () async {
```

Comment:
We need to add tests to validate all different logics, such as failed test will remove label, unauthorized reviewer will remove label, no `autosubmit` label any more will remove label, etc.

Reply:
Sure Keyong, this is also what I'm considering. I'm thinking of modifing the fake_github_service to not assign fixed values for all those methods, but allow us to give it different values when we are doing the unit test. Therefore, we can test for different situations.

Reply:
Sure Keyong, this is also the thing I'm considering. To allow test for different logics, I will remove the hardcode parts in fake_github_service and makes it flexible to parse different json data which reflects different scenarios.